ledger/eval.go

```diff
@@ -601,39 +601,34 @@ func (eval *BlockEvaluator) testTransaction(txn transactions.SignedTxn, cow *rou
 // Transaction tentatively adds a new transaction as part of this block evaluation.
 // If the transaction cannot be added to the block without violating some constraints,
 // an error is returned and the block evaluator state is unchanged.
-func (eval *BlockEvaluator) Transaction(txn transactions.SignedTxn, ad transactions.ApplyData) error {
-	return eval.transactionGroup([]transactions.SignedTxnWithAD{
-		{
-			SignedTxn: txn,
-			ApplyData: ad,
-		},
-	})
+func (eval *BlockEvaluator) Transaction(txn transactions.SignedTxn) error {
+	return eval.transactionGroup([]transactions.SignedTxn{txn})
 }
 
 // TransactionGroup tentatively adds a new transaction group as part of this block evaluation.
 // If the transaction group cannot be added to the block without violating some constraints,
 // an error is returned and the block evaluator state is unchanged.
-func (eval *BlockEvaluator) TransactionGroup(txads []transactions.SignedTxnWithAD) error {
-	return eval.transactionGroup(txads)
+func (eval *BlockEvaluator) TransactionGroup(txns []transactions.SignedTxn) error {
+	return eval.transactionGroup(txns)
 }
 
 // prepareEvalParams creates a logic.EvalParams for each ApplicationCall
 // transaction in the group
-func (eval *BlockEvaluator) prepareEvalParams(txgroup []transactions.SignedTxnWithAD) (res []*logic.EvalParams) {
+func (eval *BlockEvaluator) prepareEvalParams(txgroup []transactions.SignedTxn) (res []*logic.EvalParams) {
 	var groupNoAD []transactions.SignedTxn
 	var minTealVersion uint64
 	res = make([]*logic.EvalParams, len(txgroup))
 	for i, txn := range txgroup {
 		// Ignore any non-ApplicationCall transactions
-		if txn.SignedTxn.Txn.Type != protocol.ApplicationCallTx {
+		if txn.Txn.Type != protocol.ApplicationCallTx {
 			continue
 		}
 
 		// Initialize group without ApplyData lazily
 		if groupNoAD == nil {
 			groupNoAD = make([]transactions.SignedTxn, len(txgroup))
 			for j := range txgroup {
-				groupNoAD[j] = txgroup[j].SignedTxn
+				groupNoAD[j] = txgroup[j]
 			}
 			minTealVersion = logic.ComputeMinTealVersion(groupNoAD)
```

Comment:
I think that with your change, this entire section can now be dropped and we can use the txgroup directly in the preparation of the EvalParams below.

Reply:
must have missed this, thanks for pointing it out, fixing now